### ISSUE TITLE
Add syntax for documenting booleans: `[true]` and `[false]`

### ISSUE THREAD

**Issue by Mickeon** [godotengine/godot-proposals#5299]
### Describe the project you are working on

Taking a look around the raw Documentation files.

### Describe the problem or limitation you are having in your project

I've noticed that very, very frequently the documentation uses the following

> - _"If [code]true[/code], texture is flipped horizontally"._
> - _"If [param owned] is [code]true[/code], this method only finds nodes who have an assigned [member Node.owner]."_
> - _"If [param include_internal] is [code]false[/code], the index won't take internal children into account,"_
> - _"Used to chain two [Tweener]s after [method set_parallel] is called with [code]true[/code]."_
> - _"Returns [code]true[/code] if the [Tween] still has [Tweener]s that haven't finished."_

It's like a bit of a secret rule: whenever **true** or **false** is mentioned in a boolean context, surround it with `[code]`. Despite being everywhere, I'm surprised it doesn't have its own special syntax. Recently, parameters have received their own `[param]` syntax across the entire documentation, replacing their previously used `[code]`, as well. It opened the door for more streamlined documentation.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allow writing `[true]` and `[false]` as valid documentation, and replace the currently existing `[code]true/false[/code]`. These could display like `[code]` at first, but in the future they may have, for example, their own color, similar to the Script Editor's.

Alternatively, give some special highlighting to `true` and `false` whenever they're surrounded by code blocks.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

- _"If [param owned] is [**true**], this method only finds nodes who have an assigned [member Node.owner]."_
- _"If [param include_internal] is [**false**], the index won't take internal children into account,"_
- _"Returns [**true**] if the [Tween] still has [Tweener]s that haven't finished."_

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It cannot. It's something intrinsic to the engine.

### Is there a reason why this should be core and not an add-on in the asset library?

The documentation code is core.

**Comment by YuriSizov** [comment]
> It's like a bit of a secret rule: whenever **true** or **false** is mentioned in a boolean context, surround it with `[code]`. Despite being everywhere, I'm surprised it doesn't have its own special syntax.

There is no secret rule, it applies to all literals, be it true, false, null, numbers, quoted strings, etc. Adding special syntax for true, false, and null seems tempting, but it also doesn't bring much to the table, since there is no special case and nothing to link for them.

**Comment by Mickeon** [comment]
That's true. It's mostly for "bolder" visualization, and making it easier to read and write for the developers behind the scenes.

**Comment by dalexeev** [comment]
Another option is to use `[[x]]` as shorthand for `[code]x[/code]` (in simple cases):

```
If [param owned] is [[true]], this method only finds nodes who have an assigned [member Node.owner].
If [param include_internal] is [[false]], the index won't take internal children into account,
Returns the floating-point remainder of [[x/y]], keeping the sign of [param x].
For the integer remainder operation, use the [[%]] operator.
Equivalent of doing [[int(x)]].
```

**Comment by timothyqiu** [comment]
To be honest, I don't think this is a good idea.

Changing `[code]input[/code]` to `[param input]` clarified the semantics. But changing `[code]true[/code]` to `[true]` is merely fewer keys to type and allows for a better appearance.

Keeping BBCode compatibility as much as possible can make better use of existing linting facilities in many editors.

**Comment by Calinou** [comment]
To add to @timothyqiu's concern, remember that third-party programs reading the class reference XML (such as LSP clients) also have to interpret the new keywords to render them correctly. Most of them have `[code]` replacements in place already, but not `[true]` or `[false]`.

**Comment by Mickeon** [comment]
About that, you can make the same argument about class names which are formatted just the same way. It's not `[class AnimatedSprite2D]`, it's `[AnimatedSprite2D]`. `true`, `false`, and `null` cannot ever refer to anything else but those values. Having to refer to them is just as, if not _more_ common than a class declaration.

**Comment by AThousandShips** [comment]
But the difference is that these literals are not documented, so having `[true]` would be misleading IMO as all other tags that are `[something]` or `[type something]` are refs, not just formatting

**Comment by Mickeon** [comment]
Sorry for being stubborn on this but I seriously believe aiming for this, instead of writing down all consistent patterns presents class reference. Writing these values inside `code` tags _seriously_ is way too standardized for there not to be a shortcut of some kind.
It's the same reason as to why I heavily champion for [`[note]`, `[warning]`, `[important]` and `[tip]`](https://github.com/godotengine/godot/pull/63079), aside from the additional formatting that _these tags could benefit from_, too.

I reiterate the original feeling of this proposal, that is, it is so much more comfortable to contribute to the class reference with `[true]`, over `[code]true[/code]`.

**Comment by tetrapod00** [comment]
I'll chime in that this is clearly, self-evidently the better syntax for humans to read and write the class reference, but I'm still not sure it's worth the compatibility loss with other applications. I also quite like dalexeev's `[[literal]]` proposal, but would it not have the same problems?
